src/main/java/com/PBA/authservice/service/EmailServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.pba.authservice.service;
+
+import jakarta.mail.MessagingException;
+import jakarta.mail.internet.MimeMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class EmailServiceImpl implements EmailService {
+    @Autowired
+    private JavaMailSender mailSender;
+
+    @Override
+    public void sendVerificationEmail(String to, UUID validationCode) {
+        String subject = "Email verification";
+        String verificationLink = String.format("http://localhost:8080/api/user/%s", validationCode);
+
+        String htmlContent = String.format("""
+                <form id="verificationForm" action="%s" method="post">
+                    <input type="submit" value="Verify account">
+                </form>
+                <script>
+                    document.getElementById('verificationForm').submit();
+                </script>
+                """, verificationLink);
+        String body = String.format("""
+                Click the following button to verify your account:
```

Comment:
I appreciate the effort put in here to have also some js but **KISS**
I told you to only have a simple text where you put the url.
You can keep it as it is now since it's already implemented.